spec/modules.md

```diff
@@ -377,13 +377,19 @@ This algorithm takes a string `argument` and [configuration](#configuration)
 This algorithm takes a string, `argument`, and returns either a [source file] or
 null.
 
-* If `argument` is a relative URL:
+* If `argument` is a relative URL and the [current source file] has an
+  [importer] `importer`:
 
-  * Let `resolved` be the result of [parsing `argument` as a URL][parsing a URL]
-    with the [current source file]'s canonical URL as the base URL.
+  * If the current source file has a canonical URL `canonical`, let `url` be the
+    result of [parsing `argument` as a URL][parsing a URL] with `canonical`
+    the base URL. Otherwise, let `url` be `argument`.
```

Comment:
```suggestion
    result of [parsing `argument` as a URL][parsing a URL] with `canonical` as
    the base URL. Otherwise, let `url` be `argument`.
```